src/prop/prop_proof_manager.cpp

```diff
@@ -270,6 +270,33 @@ Node PropPfManager::normalizeAndRegister(TNode clauseNode,
 
 LazyCDProof* PropPfManager::getCnfProof() { return &d_proof; }
 
+std::vector<Node> PropPfManager::computeAuxiliaryUnits(
+    const std::vector<Node>& clauses)
+{
+  std::vector<Node> auxUnits;
+  for (const Node& c : clauses)
+  {
+    if (c.getKind() != Kind::OR)
+    {
+      continue;
+    }
+    // Determine if any OR child occurs as a top level clause. If so, it may
+    // be relevant to include this as a unit clause.
+    for (const Node& l : c)
+    {
+      const Node& atom = l.getKind() == Kind::NOT ? l[0] : l;
+      if (atom.getKind() == Kind::OR
+          && std::find(clauses.begin(), clauses.end(), atom) != clauses.end()
+          && std::find(auxUnits.begin(), auxUnits.end(), atom)
+                 == auxUnits.end())
```

Comment:
Suggestion: you could use a set for auxUnits initially, then in the end of this method build a vector out of its elements. This way you'd not be doing this `find` all the time here. I imagine this would not made a significant difference in performance either way though.

Reply:
Yes, the performance would be mixed, since `std::vector` is a lot faster if it is <15 elements or so in my experience.  I'd expect `auxUnits` to be fairly small in most cases.  I think I will keep this as std::vector for now.

Reply:
OK